superduperdb/misc/annotations.py

```diff
@@ -1,7 +1,88 @@
+import importlib
+import operator
 import sys
 import warnings
 from typing import Optional
 
+from packaging import version
+
+from superduperdb.base.exceptions import RequiredPackageNotFound
+
+ops = {
+    "<": operator.lt,
+    "<=": operator.le,
+    "==": operator.eq,
+    "!=": operator.ne,
+    ">=": operator.ge,
+    ">": operator.gt,
+}
+
+
+def _compare_versions(op, got_ver, want_ver, requirement, pkg):
+    if not ops[op](version.parse(got_ver), version.parse(want_ver)):
+        raise RequiredPackageNotFound(
+            f"{requirement} is required for a normal"
+            "functioning of this module, but found {pkg}=={got_ver}"
+        )
+
+
+def _versioned(module_name):
+    if any([o in module_name for o in ops.keys()]):
+        return True
+    return False
+
+
+def _check_versioned_module(module_name):
+    got_ver = None
+    module = module_name
+    for o in ops:
+        if o in module_name:
+            want_ver = module_name.split(o)[-1]
+            module = module_name.split(o)[0]
```

Comment:
It is better to use regex, somthing like this: 
`re.findall(r"^([^=!<>\d\.]+)([=!<>]+[\d\.]+)?$", 'transformers!=1.1.23')` or `re.findall(r"^([^=!<>\d\.]+)(([=!<>]+)([\d\.])+)?$", 'transformers!=1.1.23')`
WDYT?